Magento_Customer/templates/form/register.phtml

```diff
@@ -323,11 +329,14 @@
 <script>
     require([
         'jquery',
-        'mage/mage'
+        'mage/mage',
+        'domReady!'
     ], function($) {
         var dataForm = $('#form-validate'),
             ignore = <?= $_dob->isEnabled() ? '\'input[id$="full"]\'' : 'null' ?>;
 
+        $('._has-datepicker').datepicker('option', 'showMonthAfterYear', true);
```

Comment:
Where this element will appear? How we are sure datepicker will be already initialized?

Reply:
Datepicker will appear if $_dob->isEnabled() === true, this datepicker comes from vendor/magento/framework/View/Element/Html/Date.php `_toHtml()` method:

```
        $html .= '<script type="text/javascript">
            require(["jquery", "mage/calendar"], function($){
                    $("#' .
            $this->getId() .
            '").calendar({
                        showsTime: ' .
            ($this->getTimeFormat() ? 'true' : 'false') .
            ',
                        ' .
            ($this->getTimeFormat() ? 'timeFormat: "' .
            $this->getTimeFormat() .
            '",' : '') .
            '
                        dateFormat: "' .
            $this->getDateFormat() .
            '",
                        buttonImage: "' .
            $this->getImage() .
            '",
                        ' .
            ($calendarYearsRange ? 'yearRange: "' .
            $calendarYearsRange .
            '",' : '') .
            '
                        buttonText: "' .
            (string)new \Magento\Framework\Phrase(
                'Select Date'
            ) .
            '"' . ($maxDate ? ', maxDate: "' . $maxDate . '"' : '') .
            ($changeMonth === null ? '' : ', changeMonth: ' . $changeMonth) .
            ($changeYear === null ? '' : ', changeYear: ' . $changeYear) .
            ($showOn ? ', showOn: "' . $showOn . '"' : '') .
            ($firstDay ? ', firstDay: ' . $firstDay : '') .
            '})
            });
            </script>';
```

I've removed this line and  domReady! and just changed month-year positions using css. Let me know if it's ok for you.

Reply:
Thanks for checking it. Yes css solution is much better here.